h1b/src/usb/mod.rs

```diff
@@ -403,8 +367,8 @@ impl USB {
         // Wait until reset flag is cleared or timeout
         while self.registers.reset.get() & (Reset::CSftRst as u32) == 1 &&
             timeout > 0 {
-            timeout -= 1;
-        }
+                timeout -= 1;
+            }
```

Comment:
Would you mind reformatting this entire loop (lines 367-371)? It is confusing both before and after this change. Perhaps something like the following would be better:

```
for _ in 0..timeout {
    if self.registers.reset.get() & (Reset::CSftRst as u32) != 1 {
        continue;
    }
}
```

Reply:
Your code is cleaner, but it doesn't early exit the function if the first timeout fails. This is OK. Fixed.

Reply:
Ah yes my code is not equivalent. There's a few other ways to write the correct code instead -- e.g. I think the following is equivalent:

```
  while self.registers.reset.get() & (Reset::CSftRst as u32) == 1 {
    if timeout == 0 { return; }
    timeout -= 1;
  }
```

Reply:
Yeah, let's go with this.